language_bindings/rust/rust_build/src/lib.rs

```diff
@@ -0,0 +1,360 @@
+                .unwrap()
+        );
+        println!(
+            "file out: {:?}",
+            self.rs_output
+                .join(String::from("lib").add("runtime"))
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        fs::rename(
+            self.rs_output
+                .join("runtime")
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+            self.rs_output
+                .join(String::from("lib").add("runtime"))
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+        )
+    }
+
+    /// Renames the gen outputs to be what rust and bindgen expect on linux distros
+    ///
+    /// IE halide.a -> libhalide.a
+    pub fn rename(&self) -> Result<()> {
+        println!(
+            "file name: {:?}",
+            self.rs_output
+                .join(self.gen_name.as_str())
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        println!(
+            "file out: {:?}",
+            self.rs_output
+                .join(String::from("lib").add(self.gen_name.as_str()))
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        fs::rename(
+            self.rs_output
+                .join(self.gen_name.as_str())
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+            self.rs_output
+                .join(String::from("lib").add(self.gen_name.as_str()))
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+        )
+    }
+
+    /// Run bindgen on a generator
+    ///
+    /// This bind only creates the genname funtion binding and specificly blocklists the buffer_t
+    pub fn bind(&self) -> Result<()> {
+        let bindings = bindgen::Builder::default()
+            .header(
+                self.rs_output
+                    .join(self.gen_name.as_str())
+                    .with_extension("h")
+                    .to_str()
+                    .unwrap(),
+            )
+            .allowlist_function(self.gen_name.as_str())
+            .blocklist_item("halide_buffer_t")
+            .generate()
+            .expect("unable to gen bindings")
+            .write_to_file(
+                self.rs_output
+                    .join(self.gen_name.as_str())
+                    .with_extension("rs"),
+            );
+
+        println!("cargo:rustc-link-lib=static={}", self.gen_name);
+        bindings
+    }
+
+    /// This complies-> runs-> renames-> binds the generator
+    ///
+    /// panics if any step fails if it panics it is useful to call the functions separately and print there results or outputs
+    pub fn build_bind(&self) {
+        assert!(self.compile().status.success());
+
+        assert!(self.run_gen().status.success());
+
+        assert!(self.bind().is_ok());
```

Comment:
When a panic is desired, prefer `some_result.unwrap();` to `assert!(some_result.is_ok())`. That allows the panic message to convey some information about the failure.